index.js

```diff
@@ -171,33 +171,43 @@ async function run() {
   const ghRepo = github.getOctokit(githubToken);
 
   const commits_url = github.context.payload.pull_request.commits_url;
-  const commits = await ghRepo.request('GET ' + commits_url);
 
   const repoFullName = github.context.payload.repository.full_name;
   const repoInLicenseMap = repoFullName in licenseMap;
   var commitAuthors = {};
-  for (const commitObj of commits.data) {
-    // Check if the commit message contains a license header that matches
-    // one of the licenses granting implicit CLA approval
-    if (commitObj.commit.message && repoInLicenseMap) {
-        const goodLicense = hasImplicitLicense(commitObj.commit.message, repoFullName);
-        if (goodLicense) {
-          console.log(`- commit ${commitObj.sha} ✓ (${goodLicense} license)`);
-          continue;
+  var nCommits = 0;
+
+  await ghRepo.paginate('GET ' + commits_url,
+    {per_page: 100},
+    (response) => {
+      nCommits += response.data.length;
+      for (const commitObj of response.data) {
```

Comment:
Can we refactor this to set a `const` with commits like before? That way, cyclomatic complexity will be lower, and fewer parts of the code will be modified.

Reply:
This can be done, but it comes with painpoints.
If there are many commits to discover, they will all be stored in memory for the length of the run.
If we are good with that, then yes I can do it for sure!

Reply:
@Gu1nness memory should not be a problem, but from the code practice that's probably not ideal